azurerm/data_source_arm_scheduled_time.go

```diff
@@ -0,0 +1,172 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+type Frequency string
+
+const (
+	Day     Frequency = "Day"
+	Hour    Frequency = "Hour"
+	Month   Frequency = "Month"
+	OneTime Frequency = "OneTime"
+	Week    Frequency = "Week"
+)
+
+func dataSourceArmScheduledTime() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmScheduledTimeRead,
+		Schema: map[string]*schema.Schema{
+			"frequency": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(Day),
+					string(Hour),
+					string(Month),
+					string(OneTime),
+					string(Week),
+				}, true),
+			},
+			"second": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"minute": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"hour": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"day_of_week": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"day_of_month": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"minimum_delay_from_now_in_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  -1,
+			},
+			"timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"next_run_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmScheduledTimeRead(d *schema.ResourceData, meta interface{}) error {
+
+	var shiftTime int
+	var location *time.Location
+
+	if v, ok := d.GetOk("minimum_delay_from_now_in_minutes"); ok {
+		shiftTime = v.(int)
+	} else {
+		shiftTime = 0
+	}
+
+	if v, ok := d.GetOk("timezone"); ok {
+		var err error
+		if location, err = time.LoadLocation(v.(string)); err != nil {
+			return fmt.Errorf("Cannot parse timezone: %s", v.(string))
+		}
+	} else {
+		location = time.UTC
+	}
+
+	closestValidStartTime := time.Now().In(location).Add(time.Duration(shiftTime) * time.Minute)
+
+	var firstRunSec, firstRunMinute, firstRunHour, firstRunDayOfWeek, firstRunDayOfMonth int
+
+	//TODO: GetOk is not suitable here because it returns ok=false if the second value is 0. The GetOkExists looks good but it hasn't merged yet
```

Comment:
I'll send a PR to update the Vendored Terraform version to 0.10.6 to be able to make use of this

Reply:
PR: https://github.com/terraform-providers/terraform-provider-azurerm/pull/381

Reply:
Hi @tombuildsstuff 
sorry for the delayed answer, now i have time to complete this PR. 
Right now i am having a problem with this updated code. I can't build the code, the build fails with the error: vendor/github.com/hashicorp/terraform/config/testing.go:9: t.Helper undefined (type *testing.T has no field or method Helper).

So i am blocked now, do you have any idea what am i doing wrong?

Reply:
I just did a clean clone and now everything is working so i changed the code to use GetOkExists.

Reply:
Done!